src/engraving/dom/textbase.cpp

```diff
@@ -1868,6 +1868,21 @@ void TextBase::layoutFrame(LayoutData* ldata) const
     ldata->setBbox(ldata->frame.adjusted(-w, -w, w, w));
 }
 
+bool TextBase::positionRelativeToNoteheadRest() const
+{
+    if (!parent()) {
+        return false;
+    }
+
+    bool relativeToBarline = isMarker() || isJump() || isRehearsalMark() || isMeasureNumber() || isPlayCountText();
+    bool harmonyRelativeToBarline = isHarmony() && parent()->isFretDiagram();
+    bool textRelativeToBarline = isText() && (parent()->isBox() || parent()->isVoltaSegment());
+    bool noPositionControl = isInstrumentName() || parent()->isTuplet() || isFingering() || isHammerOnPullOffText()
+                             || parent()->isSpannerSegment();
+
+    return !relativeToBarline && !harmonyRelativeToBarline && !textRelativeToBarline && !noPositionControl;
+}
```

Comment:
Reads a bit better now, but may I suggest rewriting this in some form like:
```
switch (type()) {
case ElementType::MARKER:
case ElementType::JUMP:
case ElementType::MEASURE_NUMBER:
case ElementType::PLAY_COUNT_TEXT:
    return false;
case ElementType::HARMONY:
    return !parent()->isFretDiagram();
case ElementType::SOMETHING_ELSE:
    return somethingElse;
default:
    return basicConditions;
}
```

Cause it still reads weird to be definining a "relative to noteheade" property as "not relative to barline". Besides, those are a lot of boolean checks, a switch should be a bit more efficient (though I'm sure it hardly matters here).